File: src/ggrc/models/meeting.py

# Copyright (C) 2013 Google Inc., authors, and contributors <see AUTHORS file>
# Licensed under http://www.apache.org/licenses/LICENSE-2.0 <see LICENSE file>
# Created By:
# Maintained By:

from ggrc import db
from .mixins import Base

class Meeting(Base, db.Model):
  __tablename__ = 'meetings'

  response_id = db.Column(db.Integer, db.ForeignKey('responses.id'))
  start_at = db.Column(db.DateTime)
  calendar_url = db.Column(db.String)

  _publish_attrs = [
      'response',
      'start_at',
      'calendar_url',
      ]

File: src/ggrc/assets/javascripts/pbc/meeting.js
/*
 * Copyright (C) 2013 Google Inc., authors, and contributors <see AUTHORS file>
 * Licensed under http://www.apache.org/licenses/LICENSE-2.0 <see LICENSE file>
 * Created By:
 * Maintained By:
 */

//= require can.jquery-all
//= require models/cacheable

can.Model.Cacheable("CMS.Models.Meeting", {
  destroy : "DELETE /meetings/{id}.json"
}, {
  init : function () {
      this._super && this._super();
      // this.bind("change", function(ev, attr, how, newVal, oldVal) {
      //     var obj;
      //     if(obj = CMS.Models.ObjectDocument.findInCacheById(this.id) && attr !== "id") {
      //         obj.attr(attr, newVal);
      //     }
      // });

      var that = this;

      this.each(function(value, name) {
        if (value === null)
          that.removeAttr(name);
      });
  }

});